src/notificationmanager.cpp

```diff
@@ -69,8 +70,50 @@ NotificationManager::NotificationManager(QObject* parent)
 
 NotificationManager::~NotificationManager()
 {
+    qDeleteAll(interfaces.values());
     qDeleteAll(m_Groups);
     qDeleteAll(m_unresolvedEvents);
+
+}
+
+void NotificationManager::addModem(QString path)
+{
+    DEBUG() << "NotificationManager::addModem" << path;
+    QOfonoMessageWaiting *mw = new QOfonoMessageWaiting(this);
+    interfaces.insert(path, mw);
+
+    if (!m_messageWaiting) ///TODO: hack to use first added modem, update later to support multi-modems/sims
+	    m_messageWaiting = mw;
```

Comment:
You are right, this `m_messageWaiting` thing is clearly redundant. Are there any references to it left after your changes? `slotVoicemailWaitingChanged()` can use `sender()` to figure out which instance has sent the signal.